### PR TITLE
Support Truncated HMAC extension in ServerHello and ClientHello

### PR DESCRIPTION
**Pull request by ashfall**
https://tools.ietf.org/html/rfc6066#section-7

### CODE REVIEW

File: tls/test/test_hello_message.py [hunk 1]
```diff
@@ -245,6 +255,24 @@ def test_client_hello_maximum_fragment_length_extension_as_bytes(self):
...
+        """
+        :py:func:`tls.hello_message.ClientHello` parses a packet with a
+        truncated_hmac extension.
+        """
+        record = ClientHello.from_bytes(
+            self.client_hello_with_truncated_hmac_ext
+        )
+        assert len(record.extensions) == 1
+        assert record.extensions[0].type == enums.ExtensionType.TRUNCATED_HMAC
+        assert record.extensions[0].data == {}
```
Comment (markrwilliams) [inline tls/test/test_hello_message.py]: `Container()` instead of `{}`?

File: tls/test/test_hello_message.py [hunk 2]
```diff
@@ -422,3 +460,21 @@ def test_as_bytes_unsupported_extension(self):
...
+        """
+        :py:func:`tls.hello_message.ServerHello` parses a packet with a
+        truncated_hmac extension.
+        """
+        record = ServerHello.from_bytes(
+            self.server_hello_with_truncated_hmac_ext
+        )
+        assert len(record.extensions) == 1
+        assert record.extensions[0].type == enums.ExtensionType.TRUNCATED_HMAC
+        assert record.extensions[0].data == {}
```
Comment (markrwilliams) [inline tls/test/test_hello_message.py]: `Container()`?

File: tls/hello_message.py [hunk 3]
```diff
@@ -60,6 +60,7 @@ class ClientHello(object):
         enums.ExtensionType.SIGNATURE_ALGORITHMS,
         # XXX Incomplete list, needs to be populated as we implement more
```
Comment (markrwilliams) [inline tls/hello_message.py]: Should this comment come at the end?

Comment (ashfall) [inline tls/hello_message.py]: Yes, good catch, thanks! Updating

**Pull request merged**